### ISSUE TITLE
Exception when running in CoreCLR

### ISSUE THREAD

**Issue by StephenCleary** [xunit/aspnet.xunit#16]
Using VS2015 CTP6, with project.json of:

    {
        "version": "1.0.0-*",
        "dependencies": {
            "xunit": "2.0.0.0-rc3-build2880",
            "xunit.runner.aspnet": "2.0.0.0-rc3-build52",
            "Nito.Deque": "1.0.0-*"
        },

        "commands": {
            "test": "xunit.runner.aspnet"
        },
    
        "frameworks": {
            "aspnet50": {
                "dependencies": {
                }
            },
            "aspnetcore50": {
                "dependencies": {
                    "System.Runtime": "4.0.20-beta-22523"
                }
            }
        }
    }

Executing with the Asp.Net KRE (1.0.0-beta3 clr) works fine, but under the core CLR (1.0.0-beta3 coreclr) I get an exception:

```
PM> kvm upgrade -runtime clr
Determining latest version
kre-clr-win-x86.1.0.0-beta3 already installed.
Adding C:\Users\stephen\.k\runtimes\kre-clr-win-x86.1.0.0-beta3\bin to process PATH
Adding C:\Users\stephen\.k\runtimes\kre-clr-win-x86.1.0.0-beta3\bin to user PATH
Updating alias 'default' to 'kre-clr-win-x86.1.0.0-beta3'
PM> kvm list

Active Version     Runtime Architecture Location                     Alias  
------ -------     ------- ------------ --------                     -----  
       1.0.0-beta3 clr     x64          C:\Users\stephen\.k\runtimes        
  *    1.0.0-beta3 clr     x86          C:\Users\stephen\.k\runtimes default
       1.0.0-beta3 coreclr x64          C:\Users\stephen\.k\runtimes        
       1.0.0-beta3 coreclr x86          C:\Users\stephen\.k\runtimes        


PM> k test
xUnit.net ASP.NET test runner (32-bit Asp.Net 5.0)
Copyright (C) 2015 Outercurve Foundation.

Discovering: UnitTests
Discovered:  UnitTests
Starting:    UnitTests
Finished:    UnitTests

=== TEST EXECUTION SUMMARY ===
   UnitTests  Total: 87, Errors: 0, Failed: 0, Skipped: 0, Time: 1.301s
PM> kvm upgrade -runtime CoreCLR
Determining latest version
kre-coreclr-win-x86.1.0.0-beta3 already installed.
Adding C:\Users\stephen\.k\runtimes\kre-coreclr-win-x86.1.0.0-beta3\bin to process PATH
Adding C:\Users\stephen\.k\runtimes\kre-coreclr-win-x86.1.0.0-beta3\bin to user PATH
Updating alias 'default' to 'kre-coreclr-win-x86.1.0.0-beta3'
PM> kvm list

Active Version     Runtime Architecture Location                     Alias  
------ -------     ------- ------------ --------                     -----  
       1.0.0-beta3 clr     x64          C:\Users\stephen\.k\runtimes        
       1.0.0-beta3 clr     x86          C:\Users\stephen\.k\runtimes        
       1.0.0-beta3 coreclr x64          C:\Users\stephen\.k\runtimes        
  *    1.0.0-beta3 coreclr x86          C:\Users\stephen\.k\runtimes default


PM> k test
xUnit.net ASP.NET test runner (32-bit Asp.NetCore 5.0)
Copyright (C) 2015 Outercurve Foundation.

Discovering: UnitTests
Discovered:  UnitTests
Starting:    UnitTests
Finished:    UnitTests

=== TEST EXECUTION SUMMARY ===
k : 
At line:1 char:1
+ k test
+ ~~~~~~
    + CategoryInfo          : NotSpecified: (:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Unhandled Exception:
 
   UnitTests  Total: 0, Errors: 0, Failed: 0, Skipped: 0, Time: 0.000s
System.IO.FileLoadException: Could not load file or assembly 'System.Threading.ExecutionContext, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Could not find or load a specific file. (Exception from HRESULT: 
0x80131621) ---> System.IO.FileLoadException: Could not load the specified file.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyName(AssemblyName assemblyName)
   at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
   --- End of inner exception stack trace ---
   at Xunit.Sdk.MaxConcurrencySyncContext..ctor(Int32 maximumConcurrencyLevel)
   at Xunit.Sdk.XunitTestAssemblyRunner.SetupSyncContext(Int32 maxParallelThreads)
   at Xunit.Sdk.XunitTestAssemblyRunner.<RunTestCollectionsAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestAssemblyRunner`1.<RunAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.XunitTestFrameworkExecutor.<RunTestCases>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c__DisplayClass2.<ThrowAsync>b__5(Object state)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

I am actually wanting to run unit tests against code that uses some new .NET 4.6 APIs, so they must be run in Asp.NetCore and not Asp.Net (which is based on .NET 4.5.1).

**Comment by StefH** [comment]
I've the same issue with VS2015 CTP6 1.0.0-beta3, running tests in CoreCLR 64bit.

**Comment by bricelam** [comment]
This is probably due to the missing aspnetcore50 dependencies on the runtime packages. Try adding the following aspnetcore50 dependencies to your project.
```JSON
{
    "System.Diagnostics.Debug": "4.0.10-beta-22523",
    "System.Globalization": "4.0.10-beta-22523",
    "System.ObjectModel": "4.0.10-beta-22523",
    "System.Reflection": "4.0.10-beta-22523",
    "System.Reflection.Extensions": "4.0.0-beta-22523",
    "System.Runtime.Extensions": "4.0.10-beta-22523",
    "System.Threading.Thread": "4.0.0-beta-22523",
    "System.Threading.ThreadPool": "4.0.10-beta-22523",
    "System.Xml.XDocument": "4.0.0-beta-22523"
}
```

**Comment by StefH** [comment]
@bricelam : Sorry, that doesn't solve the issue. (Normally you don't need to add references to your project if you reference another project which does already define dependencies.

The error message looks as if a 'old' 4.0.0.0 assembly is being loaded?

```c#
System.IO.FileLoadException: Could not load file or assembly
'System.Threading.ExecutionContext, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. 
```

**Comment by bricelam** [comment]
Ah yeah, that package was removed in beta3. Its functionality now resides in `System.Threading`.

**Comment by StefH** [comment]
Probably solved by : https://github.com/xunit/xunit/pull/349 ?

**Comment by bricelam** [comment]
Probably.

**Comment by bradwilson** [comment]
Fixed in 2.1 Beta 1.
